src/main/kotlin/biokotlin/cli/ValidateAndCorrectGVCFs.kt

```diff
@@ -0,0 +1,115 @@
+
+        if (!File(referenceFile).isFile) {
+            throw IllegalArgumentException("Reference FASTA file does not exist: $referenceFile")
+        }
+
+        // Reference from FASTA file considered the correct reference sequences
+        // Map of <contig, NucSeq>
+        val refSeqGenome = fastaToNucSeq(referenceFile)
+
+        // Get list of input GVCF files from the input directory
+        val inputFiles = File(inputDir)
+            .walk()
+            .filter {
+                it.isFile && (it.name.endsWith(".g.vcf") || it.name.endsWith(".g.vcf.gz") ||
+                        it.name.endsWith(".gvcf") || it.name.endsWith(".gvcf.gz"))
+            }
+            .map { it.absolutePath }
+            .toList()
+
+        // Iterate through each GVCF file from the input directory
+        inputFiles.forEach { inputFile ->
+
+            val outputFile = "$outputDir/${File(inputFile).name}"
+            val logFile = "$outputDir/${File(inputFile).name}.log"
+
+            println()
+            println("Input GVCF file: $inputFile")
+            println("Output GVCF file: $outputFile")
+            println("Log file: $logFile")
+
+            bufferedWriter(outputFile).use { writer ->
+                bufferedWriter(logFile).use { logWriter ->
+
+                    // Write the header lines to the output GVCF file
+                    bufferedReader(inputFile).useLines { lines ->
+                        lines.takeWhile { it.startsWith("#") }.forEach { line ->
+                            writer.write("$line\n")
+                        }
+
+                        val (altHeaders, deferredVariants) = parseVCFFile(inputFile, true)
+
+                        runBlocking {
+
+                            // Iterate through each variant in the GVCF file
+                            // The work creating the deferred list is done in the parseVCFFile function
+                            // and is multithreaded
+                            for (deferred in deferredVariants) {
+                                val variant = deferred.await()
+                                val refSeq = variant.refAllele
+                                val start = variant.start
+                                val refSeqLength = refSeq.length
+                                val refSeqFromGenome =
+                                    refSeqGenome[variant.contig]?.get(start - 1 until start + refSeqLength - 1)?.seq()
+
+                                // If the reference sequence from the genome is the same as the reference sequence
+                                // in the variant, write to the output GVCF file
+                                // Otherwise, write to the log file
+                                if (refSeqFromGenome == refSeq) {
+                                    writer.write("${variant.originalText}\n")
+                                } else {
+                                    logWriter.write("Reference: $refSeqFromGenome\n")
```

Comment:
The command is named "ValidateAndCorrectGVCFs" but looks like the code just validates, with a log file presented that contains positions where the ref sequence in the genome and the vcf don't match.  Should this command just be called "ValidateGVCF" instead?  OR did I miss something?

Reply:
The writer on line 96 is the new filtered file which is the "correcting"

Reply:
But that writer is not used in this else block.  So its only writing correct seq not fixed.

Reply:
After talking with Wei-Yun, I thought those were suppose to be left out? Do you want me to replace them with the correct sequence?

Reply:
I thought we needed to fix where the MAFToGVCF was making bad calls.  It sounds like there is an Anchorwave bug which will need to be fixed.  Maybe for your command make a parameter and rename the command.  Something like this:

`biokotlin validate-gvcf --correct ...`

That way users can correct the ref errors if they would like but otherwise it validates that things are correct

Reply:
Ok, I added this